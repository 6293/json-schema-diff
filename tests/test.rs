use insta::{assert_debug_snapshot, glob, with_settings};
use json_schema_diff::diff;
use serde_json::Value;

#[test]
fn test_from_fixtures() {
    let test = |path: &std::path::Path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let values: Value = serde_json::from_str(&contents).unwrap();
        let diff = diff(values["lhs"].clone(), values["rhs"].clone()).unwrap();
        with_settings!({ info => &values }, {
            assert_debug_snapshot!(diff);
        });
    };
    glob!("../tests/fixtures", "**/*.json", test);
}
