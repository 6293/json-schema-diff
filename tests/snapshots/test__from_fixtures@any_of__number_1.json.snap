---
source: tests/test.rs
expression: diff
info:
  lhs:
    anyOf:
      - maximum: 10
        type: number
      - minimum: 1
        type: number
      - maximum: 200
        minimum: 100
        type: number
  rhs:
    anyOf:
      - maximum: 14
        minimum: 7
        type: number
      - maximum: 3
        type: number
      - minimum: 2
        type: number
input_file: tests/fixtures/any_of/number_1.json
---
[
    Change {
        path: ".<anyOf:1>",
        change: RangeChange {
            changed: Maximum,
            old_value: 10.0,
            new_value: 3.0,
        },
    },
    Change {
        path: ".<anyOf:2>",
        change: RangeChange {
            changed: Minimum,
            old_value: 1.0,
            new_value: 2.0,
        },
    },
    Change {
        path: ".<anyOf:0>",
        change: RangeChange {
            changed: Minimum,
            old_value: 100.0,
            new_value: 7.0,
        },
    },
    Change {
        path: ".<anyOf:0>",
        change: RangeChange {
            changed: Maximum,
            old_value: 200.0,
            new_value: 14.0,
        },
    },
]
